### DISCUSSION TITLE
STM32F303CCT6 with 48Kbytes of RAM

### DISCUSSION THREAD

**Discussion by psychendPerspective** [Bodmer/TFT_eSPI#1496] [Q&A - General] [answered]
Hi,
I'm looking at an application which is using the STM32F303CCT6 for a battery management system. I was wondering if the same microcontroller can also be used to interface with a 3.5/4.0 inch TFT display using this library. The readme file does suggest using STM32F1, STM32F2 or STM32F4 series, but is it worth trying with an STM32F3 series or will it absolutely not work with the RAM available? I need it to display basic graphics such as a bar graph, rpm dial and some text values. Please let me know if it will be possible or I should look at using a separate dedicated controller for interfacing with the display. 
Thank you!

**Comment by Bodmer** [comment, marked as answer]
The ram is marginal but it depends on the fonts loaded atc.

The STM32F303 has not been tested by myself so I am not sure if it will work.

My preference for a processor without WiFi is the RP2040 as it has a good performance and RAM/FLASH size and is supported.

**Comment by OscarTofa** [comment]
Hello 

I am trying to get the library to work with a nucle-32  with no success.I only get a blank screen trying the different tests

The board is a nulceo32f303k8

I think I have edited the setup file correctly by setting the following
#define ILI9341_DRIVER  
// The TFT can be connected to SPI port 1 or 2
#define TFT_SPI_PORT 1 // SPI port 1 maximum clock rate is 55MHz
#define TFT_MOSI PA7
#define TFT_MISO PA6
#define TFT_SCLK PA5
// Can use Ardiuno pin references, arbitrary allocation, TFT_eSPI controls chip select
#define TFT_CS   PA4 // Chip select control pin to TFT CS
#define TFT_DC   PB1 // Data Command control pin to TFT DC (may be labelled RS = Register Select)
#define TFT_RST  PF0 // Reset pin to TFT RST (or RESET)
I also tried with // # define ILI9341_2_DRIVER but the screen still remains blank

Is there any further tests I can do?

Thank you

**Reply by Bodmer** [reply]
You could try the generic driver by commenting out the define in your setup file thus:
```
//#define STM32
```
If it works it will be slow as then standard Arduino library functions are used.

**Reply by OscarTofa** [reply]
Hi Bodmer

yes, it's working slower, but it works correctly

**Comment by Bodmer** [comment]
OK, this indicates the low level calls used in the library are not compatible with the 303 variant.
